cluster-operator/src/main/java/io/strimzi/operator/cluster/model/dynamicconfig/KafkaBrokerConfigurationBuilder.java

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model.dynamicconfig;
+
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.strimzi.api.kafka.model.CertAndKeySecretSource;
+import io.strimzi.api.kafka.model.KafkaAuthorization;
+import io.strimzi.api.kafka.model.KafkaAuthorizationSimple;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.api.kafka.model.Rack;
+import io.strimzi.api.kafka.model.listener.KafkaListenerAuthentication;
+import io.strimzi.api.kafka.model.listener.KafkaListenerAuthenticationOAuth;
+import io.strimzi.api.kafka.model.listener.KafkaListenerAuthenticationScramSha512;
+import io.strimzi.api.kafka.model.listener.KafkaListenerAuthenticationTls;
+import io.strimzi.api.kafka.model.listener.KafkaListenerExternal;
+import io.strimzi.api.kafka.model.listener.KafkaListenerExternalIngress;
+import io.strimzi.api.kafka.model.listener.KafkaListenerExternalLoadBalancer;
+import io.strimzi.api.kafka.model.listener.KafkaListenerExternalNodePort;
+import io.strimzi.api.kafka.model.listener.KafkaListenerExternalRoute;
+import io.strimzi.api.kafka.model.listener.KafkaListeners;
+import io.strimzi.kafka.oauth.server.ServerConfig;
+import io.strimzi.operator.cluster.model.AbstractConfiguration;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+public class KafkaBrokerConfigurationBuilder {
```

Comment:
> New contributors might appreciate a sentence of Javadoc explaining what this class is for.

I added JavaDocs to all methods

> Also, as a more general point, did you consider whether we need to do something to quote user-supplied values which end up in the generated template (to prevent them being expanded incorrectly)?

I thought about it. Most of the values used here are not really freely configurable but are validated in the CRs and `api` classes. In the replacements we use are rather specific and should not be used by mistake. But TBH they also do not have any value to use them in some user configuration options. Also, when doing the `envsubst`, we allow only selected environment variables to be replaced by it and not all of them.